ibtk/src/utilities/RestartCleaner.cpp

```diff
@@ -0,0 +1,269 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2011 - 2025 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include "ibtk/IBTK_MPI.h"
+#include "ibtk/RestartCleaner.h"
+
+#include "tbox/PIO.h"
+#include "tbox/Utilities.h"
+
+#include <algorithm>
+#include <filesystem>
+#include <regex>
+
+#include "ibtk/app_namespaces.h" // IWYU pragma: keep
+
+/////////////////////////////// NAMESPACE ////////////////////////////////////
+
+namespace IBTK
+{
+/////////////////////////////// STATIC ///////////////////////////////////////
+
+/////////////////////////////// PUBLIC ///////////////////////////////////////
+
+RestartCleaner::RestartCleaner(const std::string& object_name, SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db)
+    : d_object_name(object_name),
+      d_restart_base_path(),
+      d_strategy(parseStrategy(input_db->getStringWithDefault("cleanup_strategy", "KEEP_RECENT_N"))),
+      d_keep_restart_count(input_db->getIntegerWithDefault("keep_recent_files", 5)),
+      d_enabled(input_db->getBoolWithDefault("enable_cleaner", false)),
```

Comment:
I think that this should be, in practice, the same as 'dry run', no?

I think we can implement both with `dry_run` (which should be off by default).